### ISSUE TITLE
Clarify synchronization behavior of neighborhood collectives

### ISSUE THREAD

**Issue by gcorbin** [mpi-forum/mpi-issues#863]
# Problem

The table in Appendix A.2 (Summary of the Semantics of all Operation-Related MPI Procedures) lists remark 18

> Based on their semantics, when called using an intra-communicator, MPI_ALLGATHER,
MPI_ALLTOALL, and their V and W variants, MPI_ALLREDUCE, MPI_REDUCE_SCATTER,
and MPI_REDUCE_SCATTER_BLOCK must synchronize (i.e., S1/S2 instead of W1/W2) pro-
vided that all counts and the size of all datatypes are larger than zero

also for the neighborhood collectives. Although not technically wrong(*), this remark is not relevant and confusing(**), because it does not talk about neighborhood collectives at all. 

(*):  Neighborhood collectives can be emulated by MPI_ALLTOALLW with some counts equal to zero, thus synchronization across the communicator is not guaranteed

(**): If not read carefully, could be misunderstood in the sense that synchronization is guaranteed for neighborhood collectives.

# Proposal

Remove remark 18 from all table entries belonging to neighborhood collective calls.

# Changes to the Text

Remove the remark 18) from the last column of the table in Appendix A.2, page 881 (in version 4.1) in all rows that describe neighborhood collectives. 

# Impact on Implementations

None.

# Impact on Users

None.

# References and Pull Requests

**Comment by gcorbin** [comment]
@mknobi :point_up_2:

**Comment by jeffhammond** [comment]
> it does not talk about neighborhood collectives at all.
> If not read carefully, could be misunderstood in the sense that synchronization is guaranteed for neighborhood collectives.

How can someone read so carelessly that saying nothing about neighborhood collectives leads them to conclude that synchronization is guaranteed for neighborhood collectives?

**Comment by jprotze** [comment]
> How can someone read so carelessly that saying nothing about neighborhood collectives leads them to conclude that synchronization is guaranteed for neighborhood collectives?

Probably by trusting in the mpi forum, and that these folks have a reason to explicitly reference the remark for the group of blocking neighborhood collectives.

I agree with @gcorbin , the mark should be removed from those rows unless there is a reason, in which case the text of the remark would need to be extended to cover neighborhood collectives.

**Comment by jeffhammond** [comment]
In my experience, the only people who read the standard that carefully are in the MPI Forum.

I don't object to removing this text if it's redundant, but I object to removing it because somebody might be confused about a topic where we say nothing.

**Comment by mknobi** [comment]
This remark is made for multiple items in the table in appendix A2
- For the blocking collective routines and the corresponding tests and waits for the non-blocking variants (p 880, lines 4, 15, 25) -> correct 
- For the blocking neighborhood collective routines and the corresponding tests and waits for the non-blocking variants (p 881, lines 7, 15, 22, 23) -> pointless, as this remark is not about neighborhood collectives (and doesn't apply to them, as @gcorbin pointed out).

As such, all occurrences on page 881 should be removed

**Comment by wesbland** [comment]
Obviously this issue is too late to be discussed in today's meeting. If it's something we really want to get into MPI-next, we can put it on the list for the March meeting.

**Comment by mknobi** [comment]
Yes, it makes sense to discuss this in Stuttgart. As @jeffhammond pointed out, this is an issue that won't affect most users, so this is not urgent.

**Comment by RolfRabenseifner** [comment]
The proposal _"Remove remark 18 from all table entries belonging to neighborhood collective calls"_ is correct. Remark 18) for neighborhood collectives is wrong, because in the case of a (dist) graph topology with not connected subgraphs, remark 18 is really wrong. Therefore, please remove all Five "18)" on MPI-4.1 page 881 (printed page number).
My apologies because I expect that I added this remark to these routines. And thanks to Christoph that he asked me about my opinion on this issue.

**Comment by wesbland** [comment]
This passed a [errata vote](https://mpi-forum.org/meetings/2025/09/votes).
<table>
<tr>
    <td> Yes
    <td> No
    <td> Abstain
<tr>
    <td> 28
    <td> 1
    <td> 3
</table>
